2017_01_04_word_count/wc.js

```diff
@@ -0,0 +1,23 @@
+// Run this with:
+// node wc.js quotes.txt
+var fileName = process.argv[2];
+fs = require('fs')
+fs.readFile(fileName, 'utf8', function (err, data) {
+    if (err) {
+        return console.log(err);
+    }
+
+    console.log(lineCount(data) + ' ' + wordCount(data) + ' ' + totalChars(data) + ' ' + fileName);
+});
+
+function lineCount(data) {
```

Comment:
Could you help me out so I can understand this better? Is there an article about what cases would it matter? I hadn't heard that before. My understanding was that function hoisting would ensure no issues here.

Hi Clay, I'm Britton by the way.